Barrier package failing to build

### Describe the bug
The `barrier` is failing to build

### Steps To Reproduce
Steps to reproduce the behavior:
1. Run `nix build github:nixos/nixpkgs/master#barrier`
```
...
[ 12%] Building CXX object src/lib/base/CMakeFiles/base.dir/Event.cpp.o
In file included from /build/source/src/./lib/base/EventTypes.h:20,
                 from /build/source/src/./lib/io/StreamBuffer.h:21,
                 from /build/source/src/lib/io/StreamBuffer.cpp:19:
/build/source/src/./lib/base/Event.h:63:37: error: 'NULL' was not declared in this scope
   63 |     Event(Type type, void* target = NULL, void* data = NULL,
      |                                     ^~~~
/build/source/src/./lib/base/Event.h:23:1: note: 'NULL' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
   22 | #include "common/stdmap.h"
  +++ |+#include <cstddef>
   23 |
/build/source/src/./lib/base/Event.h:63:56: error: 'NULL' was not declared in this scope
   63 |     Event(Type type, void* target = NULL, void* data = NULL,
      |                                                        ^~~~
/build/source/src/./lib/base/Event.h:63:56: note: 'NULL' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
[ 12%] Building CXX object src/lib/mt/CMakeFiles/mt.dir/Lock.cpp.o
/build/source/src/lib/io/StreamBuffer.cpp: In member function 'const void* StreamBuffer::peek(UInt32)':
/build/source/src/lib/io/StreamBuffer.cpp:49:16: error: 'NULL' was not declared in this scope
   49 |         return NULL;
      |                ^~~~
/build/source/src/lib/io/StreamBuffer.cpp:26:1: note: 'NULL' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
   25 | #include <cassert>
  +++ |+#include <cstddef>
   26 |
make[2]: *** [src/lib/io/CMakeFiles/io.dir/build.make:76: src/lib/io/CMakeFiles/io.dir/StreamBuffer.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:577: src/lib/io/CMakeFiles/io.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 12%] Building CXX object src/lib/mt/CMakeFiles/mt.dir/Mutex.cpp.o
In file included from /build/source/src/lib/base/Event.cpp:19:
/build/source/src/./lib/base/Event.h:63:37: error: 'NULL' was not declared in this scope
   63 |     Event(Type type, void* target = NULL, void* data = NULL,
      |                                     ^~~~
/build/source/src/./lib/base/Event.h:23:1: note: 'NULL' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
   22 | #include "common/stdmap.h"
  +++ |+#include <cstddef>
   23 |
/build/source/src/./lib/base/Event.h:63:56: error: 'NULL' was not declared in this scope
   63 |     Event(Type type, void* target = NULL, void* data = NULL,
      |                                                        ^~~~
/build/source/src/./lib/base/Event.h:63:56: note: 'NULL' is defined in header '<cstddef>'; did you forget to '#include <cstddef>'?
[ 13%] Building CXX object src/lib/mt/CMakeFiles/mt.dir/Thread.cpp.o
[ 13%] Building CXX object src/lib/mt/CMakeFiles/mt.dir/XMT.cpp.o
make[2]: *** [src/lib/base/CMakeFiles/base.dir/build.make:76: src/lib/base/CMakeFiles/base.dir/Event.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:499: src/lib/base/CMakeFiles/base.dir/all] Error 2
[ 14%] Linking CXX static library libmt.a
[ 14%] Built target mt
make: *** [Makefile:166: all] Error 2
```

### Expected behavior
The package to build successfully without errors

### Notify maintainers
@phryneas

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.17.1, NixOS, 22.05 (Quokka), 22.05.20220417.1ffba9f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.9.0pre20220422_7f814d6`
```

--- a/pkgs/applications/misc/barrier/default.nix
+++ b/pkgs/applications/misc/barrier/default.nix
@@ -1,6 +1,7 @@
 { lib, fetchFromGitHub, cmake, curl, xorg, avahi, qtbase, mkDerivation,
   openssl, wrapGAppsHook,
-  avahiWithLibdnssdCompat ? avahi.override { withLibdnssdCompat = true; }
+  avahiWithLibdnssdCompat ? avahi.override { withLibdnssdCompat = true; },
+  fetchpatch
 }:
 
 mkDerivation rec {
@@ -14,6 +15,16 @@ mkDerivation rec {
     sha256 = "sha256-2tHqLF3zS3C4UnOVIZfpcuzaemC9++nC7lXgFnFSfKU=";
     fetchSubmodules = true;
   };
+
+  patches = [
+    # This patch can be removed when a new version of barrier (greater than 2.4.0)
+    # is released, which will contain this commit.
+    (fetchpatch {
+      name = "add-missing-cstddef-header.patch";
+      url = "https://github.com/debauchee/barrier/commit/4b12265ae5d324b942698a3177e1d8b1749414d7.patch";
+      sha256 = "sha256-ajMxP7szBFi4h8cMT3qswfa3k/QiJ1FGI3q9fkCFQQk=";
+    })
+  ];
 
   buildInputs = [ curl xorg.libX11 xorg.libXext xorg.libXtst avahiWithLibdnssdCompat qtbase ];
   nativeBuildInputs = [ cmake wrapGAppsHook ];

